mldsa/polyvec.c

```diff
@@ -11,14 +11,38 @@
 void polyvec_matrix_expand(polyvecl mat[MLDSA_K],
                            const uint8_t rho[MLDSA_SEEDBYTES])
 {
-  unsigned int i, j;
+  /* Reference: This code is re-factored from the reference implementation
+   * to facilitate proof with CBMC and to improve readability.
+   *
+   * The temporary polyvecl and poly variables simplify the CBMC
+   * memory model by removing nested access patterns. Loop invariant
+   * checks are performed on the local variable tmp_polyvecl before
+   * assigning it to the output matrix. */
+
+  unsigned int i;
 
   for (i = 0; i < MLDSA_K; ++i)
+  __loop__(
+    assigns(i, memory_slice(mat, MLDSA_K * sizeof(polyvecl)))
+    invariant(i <= MLDSA_K)
+    invariant(forall(k1, 0, i, forall(l1, 0, MLDSA_L, array_bound(mat[k1].vec[l1].coeffs, 0, MLDSA_N, 0, MLDSA_Q))))
+  )
   {
+    unsigned int j;
+    polyvecl tmp_polyvecl;
```

Comment:
Is this a temporary workaround for https://github.com/diffblue/cbmc/issues/8617? If so, please add an appropriate comment with a TODO.